### PR TITLE
Add uploading to asset server event

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3672/HighFidelity-Beta-PR11451-88810d28c1f39c19ffc1d5f4a29aa0fdcf3d0afd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3697/HighFidelity-Beta-PR11451-0e3c2f2e50ca12a569dd9c1ba619fda44fcb5fd8.exe

**Comment by huffman** [comment]
test this please

**Comment by huffman** [comment]
To test:

 1. Go to localhost.
 1. Remove everything from your domain.
 1. Restart interface and go to localhost
 1. Upload https://hifi-public.s3.amazonaws.com/ozan/sets/secretGardenParty/secretGardenParty.fbx as a new fbx to your Asset Server.
 1. Add that asset to your domain
 1. Add `https://hifi-public.s3.amazonaws.com/ozan/sets/secretGardenParty/secretGardenParty.fbx` to your domain
 1. Wait for everything to load

Please let me know once you have done all of that.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3714/HighFidelity-Beta-PR11451-0e3c2f2e50ca12a569dd9c1ba619fda44fcb5fd8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3714/HighFidelity-Beta-PR11451-0e3c2f2e50ca12a569dd9c1ba619fda44fcb5fd8.dmg

**Comment by OmegaHeron** [comment]
Will test.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3725/HighFidelity-Beta-PR11451-97bf093cabda4dfcb98d2c1416b122533ef8898b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3725/HighFidelity-Beta-PR11451-97bf093cabda4dfcb98d2c1416b122533ef8898b.exe

**Comment by OmegaHeron** [comment]
@huffman - Test complete.

**Comment by huffman** [comment]
Looks great, thanks @OmegaHeron !

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 1]
```diff
@@ -104,6 +105,25 @@ void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping,
...
+    QFile file { path };
+    int64_t size { 0 };
+    if (file.open(QIODevice::ReadOnly)) {
+        size = file.size();
+        file.close();
+    }
+
+    QString extension = "";
+    auto idx = path.lastIndexOf(".");
+    if (idx) {
```
Comment (huffman) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp]: I need to fix this to be `>= 0`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3739/HighFidelity-Beta-PR11451-ff019d619516cb1e9b91006c311ff0bdca7a3730.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3739/HighFidelity-Beta-PR11451-ff019d619516cb1e9b91006c311ff0bdca7a3730.exe

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 2]
```diff
@@ -104,6 +105,25 @@ void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping,
 
     startedCallback.call();
 
+    QFile file { path };
+    int64_t size { 0 };
+    if (file.open(QIODevice::ReadOnly)) {
+        size = file.size();
+        file.close();
+    }
```
Comment (Atlante45) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp]: Can you use `QFileInfo::size` instead? That won't require us to open the file.

Comment (huffman) [reply interface/src/scripting/AssetMappingsScriptingInterface.cpp]: Ah, yes, I'll do that.

Comment (huffman) [reply interface/src/scripting/AssetMappingsScriptingInterface.cpp]: Thanks, updated.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4002/HighFidelity-Beta-PR11451-9c120f1194e9994abbfe89257de16466e3485440.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4002/HighFidelity-Beta-PR11451-9c120f1194e9994abbfe89257de16466e3485440.exe

**Pull request merged**